inc/XamlApplication.h

```diff
@@ -105,4 +107,47 @@ namespace winrt {
   {
     return winrt::make<winrt::CppXaml::implementation::XamlApplication>(std::initializer_list<winrt::Windows::UI::Xaml::Markup::IXamlMetadataProvider>{providers...});
   }
+}
+
+#pragma comment(lib, "onecoreuap_apiset.lib")
+
+namespace cppxaml {
+  namespace details {
+    PWSTR g_winUIDepId{};
+    PACKAGEDEPENDENCY_CONTEXT g_ctx{};
+  }
+
+  /// <summary>
+  /// Dynamically initializes WinUI 2. This API works on Windows 11 and newer.
+  /// Adds the WinUI 2 framework package to the process's package graph.
+  /// This enables unpackaged applications to use the stable (non-prerelease) WinUI 2 NuGet package.
+  /// </summary>
+  /// <param name="minorVersion">Optional minor version. Defaults to 8 (for 2.8)</param>
+  void InitializeWinUI(uint8_t minorVersion = 8) {
+
+    if (details::g_winUIDepId == nullptr) {
+      PACKAGE_VERSION minVersion{};
+      minVersion.Major = minorVersion;
+
+      constexpr const std::wstring_view winuiPackageFamilyNameTemplate{ L"Microsoft.UI.Xaml.2.%d_8wekyb3d8bbwe" };
+      wchar_t pfn[PACKAGE_FAMILY_NAME_MAX_LENGTH]{};
+      winrt::check_hresult(StringCchPrintf(pfn, std::size(pfn), winuiPackageFamilyNameTemplate.data(), minorVersion));
+
+      winrt::check_hresult(TryCreatePackageDependency(nullptr, pfn, minVersion,
+        PackageDependencyProcessorArchitectures_None, PackageDependencyLifetimeKind_Process,
+        nullptr, CreatePackageDependencyOptions_None, &details::g_winUIDepId));
+    }
+
+    PWSTR packageFullName{};
+
+    winrt::check_hresult(AddPackageDependency(details::g_winUIDepId, 0, AddPackageDependencyOptions_None, &details::g_ctx, &packageFullName));

```

Comment:
The documentation for AddPackageDependency mentions that `packageFullName` must be manually freed:
> The caller is responsible for freeing this resource once it is no longer needed by calling [HeapFree](https://learn.microsoft.com/en-us/windows/win32/api/heapapi/nf-heapapi-heapfree).